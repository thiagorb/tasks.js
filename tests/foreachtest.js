var Task = require("Task");

var data = [
    {
        name: "colors",
        values: ["red", "green", "blue"]
    },
    {
        name: "nothing",
        values: []
    },
    {
        name: "numbers",
        values: ["one", "two", "three", "four"]
    }
];

Task.ForEach(data, function (flow, item) {
    flow.suspend();
    console.log(item.name + ":");
    Task.ForEach(item.values, function (flow, value) {
        flow.suspend();
        console.log("-" + value);
        setTimeout(function () { flow.resume(); }, 10);
    }).then(function () {
        flow.resume();
    }).start();
}).start();