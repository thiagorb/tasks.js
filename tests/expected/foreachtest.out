colors:
-red
-green
-blue
nothing:
numbers:
-one
-two
-three
-four